jujugui/static/gui/src/app/app.js

```diff
@@ -2110,6 +2109,21 @@ YUI.add('juju-gui', function(Y) {
       next();
     },
 
+    _ensureControllerConnection: function(state, next) {
+      if (
+        state.root === 'logout' ||
+        this.controllerAPI.get('connecting') ||
+        this.controllerAPI.get('connected') ||
+        this.controllerAPI.userIsAuthenticated) {
```

Comment:
I don't think this is required since the above `connecting` and `connected` will cover pretty much every case.